### PR TITLE
DDBTEAM-636: Added new source download command

### PR DESCRIPTION
**Pull request by cableman**

### CODE REVIEW

File: src/Command/SearchPopulateCommand.php [hunk 1]
```diff
@@ -62,6 +62,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $this->populateService->setProgressBar($progressBar);
         $this->populateService->populate($index, $id);
 
+        $output->writeln('');
```
Comment (turegjorup) [inline src/Command/SearchPopulateCommand.php:65]: What is the purpose of this? Either remove or explain in comment

File: src/Command/Source/SourceDownloadCoversCommand.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $vendorId = $input->getOption('vendor-id');
+        $identifier = $input->getOption('identifier');
+
+        $batchSize = 50;
```
Comment (turegjorup) [inline src/Command/Source/SourceDownloadCoversCommand.php:61]: Would prefer a class constant, but given that it is just a command it's not important

File: src/Command/Source/SourceDownloadCoversCommand.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        $queryStr = 'SELECT s FROM App\Entity\Source s WHERE s.image IS NULL AND s.originalFile IS NOT NULL';
+        if (!is_null($identifier)) {
+            $queryStr .= ' AND s.matchId = '.$identifier;
+        }
+        if (!is_null($vendorId)) {
+            $queryStr .= ' AND s.vendor = '.$vendorId;
+        }
+
+        $query = $this->em->createQuery($queryStr);
+        $iterableResult = $query->iterate();
```
Comment (turegjorup) [inline src/Command/Source/SourceDownloadCoversCommand.php:80]: Have you tested this in regards to memory consumption. I believe we have had memory issues with the doctrine iterator before...

**Review by turegjorup — APPROVED** [review]
Approved with comments

File: src/Command/Source/SourceDownloadCoversCommand.php (hunk 3 above)
Comment (cableman) [reply src/Command/Source/SourceDownloadCoversCommand.php:80]: Yes it's slowly goes up...

Comment (cableman) [reply src/Command/Source/SourceDownloadCoversCommand.php:80]: But we use it in other commands... and there is a clear/flush for each batch

File: src/Command/Source/SourceDownloadCoversCommand.php (hunk 2 above)
Comment (cableman) [reply src/Command/Source/SourceDownloadCoversCommand.php:61]: `Constants defined using the const keyword must be declared at the top-level scope`

File: src/Command/SearchPopulateCommand.php (hunk 1 above)
Comment (cableman) [reply src/Command/SearchPopulateCommand.php:65]: Comment added

File: src/Command/Source/SourceDownloadCoversCommand.php (hunk 2 above)
Comment (cableman) [reply src/Command/Source/SourceDownloadCoversCommand.php:61]: Don't what it in top level.

File: src/Command/Source/SourceDownloadCoversCommand.php (hunk 3 above)
Comment (turegjorup) [reply src/Command/Source/SourceDownloadCoversCommand.php:80]: 👍

File: src/Command/Source/SourceDownloadCoversCommand.php (hunk 2 above)
Comment (turegjorup) [reply src/Command/Source/SourceDownloadCoversCommand.php:61]: 👍

**Review by turegjorup — APPROVED** [review]
Approved, but please add the two missing doc comments

**Pull request merged**